### PR TITLE
Implement contactListMapFromJson() and contactListMapToJson() methods

### LINKED ISSUE #315: Implement a class that reads the planned footsteps from file
Discussing with @paolo-viceconte we realized that it would be nice to have a method/class that reads from a file the position and timings of the footsteps. A possible solution is to use a JSON file (already used by @paolo-viceconte in python)

Here a library that can be used to parse a JSON file https://github.com/nlohmann/json

The library can be installed from apt/vcpkg/brew/conda

cc @traversaro @S-Dafarra

### PR DESCRIPTION
**Pull request by GiulioRomualdi**
This PR closes #315 in detail I used [nlohmann json](https://github.com/nlohmann/json/) to parse the JSON file.

The two functions can be used to store and read from a JSON file the contact lists.

TODO
- [x]  write the tests
- [x] Update the CI installing  [nlohmann json](https://github.com/nlohmann/json/) (At least 3.7.3 is required - the one provided by `apt`)

### CODE REVIEW

File: src/Contacts/include/BipedalLocomotion/Contacts/ContactListParser.h [hunk 1]
```diff
@@ -0,0 +1,39 @@
+/**
+ * @file ContactListParser.h
+ * @authors Giulio Romualdi
+ * @copyright 2021 Istituto Italiano di Tecnologia (IIT). This software may be modified and
+ * distributed under the terms of the GNU Lesser General Public License v2.1 or any later version.
+ */
+
+#ifndef BIPEDAL_LOCOMOTION_CONTACTS_CONTACTLIST_PARSER_H
+#define BIPEDAL_LOCOMOTION_CONTACTS_CONTACTLIST_PARSER_H
```
Comment (S-Dafarra) [inline src/Contacts/include/BipedalLocomotion/Contacts/ContactListParser.h:9]: I don't know if it is worth adding "File" or "Json" also in the name of this file.

File: src/Contacts/src/ContactListParser.cpp [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+                             logPrefix,
+                             contactPosition.size(),
+                             contactQuaternion.size());
+                return ContactListMap();
+            }
+
+            tempContact.pose.quat(Eigen::Map<const manif::SO3d>(contactQuaternion.data()));
+            tempContact.pose.translation(Eigen::Map<const manif::R3d>(contactPosition.data()).coeffs());
+            tempContact.activationTime = val.at("activation_time");
+            tempContact.deactivationTime = val.at("deactivation_time");
```
Comment (S-Dafarra) [inline src/Contacts/src/ContactListParser.cpp:67]: Maybe it is worth adding somewhere the expected format

File: src/Contacts/include/BipedalLocomotion/Contacts/ContactListParser.h (hunk 1 above)
Comment (GiulioRomualdi) [reply src/Contacts/include/BipedalLocomotion/Contacts/ContactListParser.h]: File renamed in `ContactListJsonParser.h`

File: src/Contacts/src/ContactListParser.cpp (hunk 2 above)
Comment (GiulioRomualdi) [reply src/Contacts/src/ContactListParser.cpp]: Done in https://github.com/dic-iit/bipedal-locomotion-framework/pull/316/commits/b7be6c963ca04999a14e4a46aaf752862463ed9e

File: bindings/python/Contacts/src/Contacts.cpp [hunk 3]
```diff
@@ -170,6 +171,19 @@ void CreateContactPhaseList(pybind11::module& module)
         .def("lists", &ContactPhaseList::lists);
 }
 
+void CreateContactParser(pybind11::module& module)
```
[the remark this answers is not in the document]
Comment (GiulioRomualdi) [reply bindings/python/Contacts/src/Contacts.cpp]: Here https://github.com/dic-iit/bipedal-locomotion-framework/pull/316/commits/e315e512497ecf31d3846e4df64cac841f42c705

**Comment by GiulioRomualdi** [comment]
@traversaro before merging (or just after) the PR we should add:
- in the superbuild the guide to install  nlohmann json
- in vcpkg-robotolgy  the installation of  nlohmann json

**Comment by GiulioRomualdi** [comment]
@S-Dafarra if you agree I can merge the PR

**Review by S-Dafarra — COMMENTED** [review]
Sorry, I somehow lost this

**Review by paolo-viceconte — APPROVED** [review]
It looks good to me. 

As a doublecheck, I also tried to store a sequence of footsteps from hadherent through `contact_list_map_to_json()`, which then @GiulioRomualdi was able to read.

File: src/Contacts/include/BipedalLocomotion/Contacts/ContactListJsonParser.h [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+namespace BipedalLocomotion
+{
+namespace Contacts
+{
+
+/**
+ * Parse a ContactListMap from a JSON file.
+ * @param filename the name of the file that should be loaded.
+ * @return return a ContactListMap containing all the contacts.
+ * @note The JSON file should have the following structure
```
[the remark this answers is not in the document]
Comment (GiulioRomualdi) [reply src/Contacts/include/BipedalLocomotion/Contacts/ContactListJsonParser.h]: Here I did some modifications https://github.com/dic-iit/bipedal-locomotion-framework/pull/316/commits/f14c9d638cc0d564496cbdca928290d1be697001

what do you think?

**Review by S-Dafarra — APPROVED** [review]

File: src/Contacts/include/BipedalLocomotion/Contacts/ContactListJsonParser.h (hunk 4 above)
[the remark this answers is not in the document]
Comment (S-Dafarra) [reply src/Contacts/include/BipedalLocomotion/Contacts/ContactListJsonParser.h]: Looks good to me!

**Pull request merged**